lib/kernel.coffee

```diff
@@ -145,15 +145,22 @@ class Kernel
 
         @signedSend message, @shellSocket
 
-    execute: (code, onResults) ->
+    handleExtensions: (code, language) ->
+        if language == 'coffeescript'
+            if language not in @loadedExtensions
+                @_execute '%load_ext coffee', => @loadedeExtensions.push language
+            "%%coffee bare no-prompt\n"+code
```

Comment:
My main issue with this is that it flies in the face of the JavaScript kernel I'm currently using, [ijavascript](https://github.com/n-riesco/ijavascript). It would be nice to [unify the JS/Node kernels](https://github.com/notablemind/jupyter-nodejs/issues/4), otherwise we're going to have this special matrix of handlers for Hydrogen.

Then again, if there's a way we can tell which kernel is setup, we can defer to the right one.